src/core/TrainSystem.hpp

```diff
@@ -1,19 +1,18 @@
 #pragma once
-
+#include "./LogSystem.hpp"
 #include <godot_cpp/classes/engine.hpp>
 #include <godot_cpp/classes/ref_counted.hpp>
 #include <godot_cpp/core/class_db.hpp>
 #include <godot_cpp/variant/array.hpp>
 #include <godot_cpp/variant/string.hpp>
 #include <godot_cpp/variant/utility_functions.hpp>
 #include <map>
-#include "./LogSystem.hpp"
 
 namespace godot {
 
     class TrainController;
 
-    class TrainSystem : public RefCounted {
+    class TrainSystem final : public RefCounted {
```

Comment:
I don't see the point of "final" here. It could be added, of course, because it changes nothing for now.